Jropp/jason-ropp-js-iobp
# Onboarding Project: Move Icon Toggle Into Dom-If Templates

## What It Does

This moves the display switching of icons into a dom-if template. 

Previously I was toggling the icons by changing the inner html of the createUserButton. This was a lot of lines of extra code. Moving the display toggle into the dom-if template allowed me to use existing classes and displays to set the display. 

Both for the expand/collapse details up/down arrow in the user card, and for the add/cancel new user icons, I put them both into dom-if templates and used the isClass() function for the boolean:

```js
        <button id="addUserButton" on-click="toggleNewUserCardDisplay" class="create-user-button">

            <template is="dom-if" if="[[isClass(classes.COLLAPSE_NEW, userCardClass)]]">
              <jha-add-person-icon></jha-add-person-icon>
            </template>

            <template is="dom-if" if="[[isClass(classes.EXPAND_NEW, userCardClass)]]">
              <jha-cancel-icon></jha-cancel-icon>
            </template>

          </button>
```

## How To Test

- Opening and closing details in a user card should toggle the direction of the arrow appropriately.
- Clicking the `add new user` or `cancel` icon should toggle appropriately as well. 
- Try a random way to break the code outside of the scope of this PR, some odd combination of entries or deletes.

## Notes/Caveats

## Checklist

Under penalty of public shaming, I avow that I:


Browsers tested:

## Dependencies

- [ ] #99
